docs/transpile/custom-transpiler-pass.ipynb

```diff
@@ -0,0 +1,239 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Transpiler Passes\n",
+    "\n",
+    "Transpiler passes are classified into one of two classes: Analysis passes and Transformation passes. Passes in general work with the Directed Acyclic Graph (DAG) **TODO: insert link here** and the `property_set`, a dictionary-like object for storing properties determined by analysis passes. Analysis passes work with the DAG, but are not allowed to modify it. This contrasts with transformation passes, which do modify the DAG, and can read (but not write to) `property_set`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Internals\n",
+    "\n",
+    "**TODO: put something here, but I do not know what**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Pauli Twirling\n",
```

Comment:
```suggestion
    "## Create a `PauliTwirl` transpiler pass\n",
```
I think somewhere it should be highlighted what the actual name of your transpiler pass is. I can kind of figure it out from the code, but I think it would be good to call it out in the main text as well